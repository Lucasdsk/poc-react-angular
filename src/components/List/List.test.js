import React from "react";
import { List } from "./index";

describe("<List />", () => {
  it("Render Component", () => {
    const listItems = [
      { id: 1, name: "Test", age: 20, email: "test@ambev.com.br" },
      { id: 2, name: "Test1", age: 21, email: "test1@ambev.com.br" }
    ];
    const ListComponent = shallow(<List items={listItems} />);

    expect(ListComponent).toMatchSnapshot();
  });

  it("Should have the corrects props", () => {
    const listItems = [
      { id: 1, name: "Test", age: 20, email: "test@ambev.com.br" },
      { id: 2, name: "Test1", age: 21, email: "test1@ambev.com.br" }
    ];
    const handleRemoveItem = jest.fn();
    const props = {
      items: listItems,
      onRemoveItem: handleRemoveItem
    };
    const ListComponent = mount(
      <List items={listItems} onRemoveItem={handleRemoveItem} />
    );

    expect(ListComponent.props()).toEqual(props);
  });

  it("Should render message when doesn't have items", () => {
    const listItems = [];
    const ListComponent = shallow(<List items={listItems} />);

    expect(ListComponent.find("h3")).toHaveLength(1);
  });

  it("Should render items", () => {
    const listItems = [
      { id: 1, name: "Test", age: 20, email: "test@ambev.com.br" },
      { id: 2, name: "Test1", age: 21, email: "test1@ambev.com.br" },
      { id: 3, name: "Test2", age: 22, email: "test2@ambev.com.br" }
    ];
    const ListComponent = shallow(<List items={listItems} />);

    expect(ListComponent.find(".list")).toHaveLength(1);
    expect(ListComponent.find(".list__item")).toHaveLength(3);
  });

  it("Should remove item when user clicks on remove's button", () => {
    const listItems = [
      { id: 1, name: "Test", age: 20, email: "test@ambev.com.br" },
      { id: 2, name: "Test1", age: 21, email: "test1@ambev.com.br" }
    ];
    const handleRemoveItem = jest.fn();
    const ListComponent = shallow(
      <List items={listItems} onRemoveItem={handleRemoveItem} />
    );
    const listItem = ListComponent.find(".list__item")
      .first()
      .find(".list__item__remove-item");
      
    listItem.simulate("click");
    expect(handleRemoveItem).toHaveBeenCalled();
  });
});
