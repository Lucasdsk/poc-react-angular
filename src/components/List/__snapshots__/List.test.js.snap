// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> Render Component 1`] = `
<styled.div>
  <ul
    className="list"
  >
    <li
      className="list__item"
      key="1"
    >
      <div
        className="list__item__data"
      >
        <div>
          Test
        </div>
        <div>
          20
        </div>
        <div>
          test@ambev.com.br
        </div>
      </div>
      <span
        className="list__item__remove-item"
        onClick={[Function]}
      >
        x
      </span>
    </li>
    <li
      className="list__item"
      key="2"
    >
      <div
        className="list__item__data"
      >
        <div>
          Test1
        </div>
        <div>
          21
        </div>
        <div>
          test1@ambev.com.br
        </div>
      </div>
      <span
        className="list__item__remove-item"
        onClick={[Function]}
      >
        x
      </span>
    </li>
  </ul>
</styled.div>
`;
